api/client/action/pruner.go

```diff
@@ -17,13 +16,12 @@ const apiName = "ActionPruner"
 // Facade allows calls to "ActionPruner" endpoints.
 type Facade struct {
 	facade base.FacadeCaller
-	*common.ModelWatcher
 }
 
 // NewPruner builds a facade for the action pruner endpoints.
 func NewPruner(caller base.APICaller) *Facade {
 	facadeCaller := base.NewFacadeCaller(caller, apiName)
-	return &Facade{facade: facadeCaller, ModelWatcher: common.NewModelWatcher(facadeCaller)}
```

Comment:
So this pruner is in the client package but is supposed to be controller only. If it turns out to be client facade you will need to bump the version to ensure this works and ensure that model-migration also behaves itself. I'd probably do the migrate from 3.6 to main anyway.

Reply:
I think this is just a matter of it being placed in the wrong directory. It was added in the same package as the action facade client, even though the only usage I can see of this facade in both 3.6 and main is in the actionpruner worker.

I'll test with 3.6 just to be safe.

Reply:
Agreed, but you can never be too sure.

Thanks for doing the test.

Reply:
If it is fine, can we then move this to `api/controller/action` package.

Reply:
So:
- I moved the actionpruner client to the `api/controller` package.
- I tested all the QA steps with a 3.6 client and 4.0 controller, all worked fine
- I also tested a 3.6 to 4.0 model migration.